middleware/parse/parse_test.go

```diff
@@ -146,6 +183,72 @@ func TestParseMiddleware(t *testing.T) {
 		assert.Equal(t, http.StatusOK, result.StatusCode)
 	})
 
+	t.Run("Invalid Multipart", func(t *testing.T) {
+		// Write empty body, which is not allowed for content multipart.
+		writer := multipart.NewWriter(nil)
+
+		request := testutil.NewTestRequest(http.MethodPost, "/parse", nil)
+		request.Lang = server.Lang.GetDefault()
+		request.Header().Set("Content-Type", writer.FormDataContentType())
+
+		result := server.TestMiddleware(&Middleware{}, request, func(resp *goyave.Response, _ *goyave.Request) {
+			resp.Status(http.StatusBadRequest)
+		})
+
+		assert.NoError(t, result.Body.Close())
+		assert.Equal(t, http.StatusBadRequest, result.StatusCode)
+		assert.NotNil(t, request.Extra[goyave.ExtraParseError{}])
+		assert.NotPanics(t, func() {
+			extraError, ok := request.Extra[goyave.ExtraParseError{}].(error)
+			require.True(t, ok)
+			assert.ErrorIs(t, extraError, goyave.ErrInvalidContentForType)
+		})
+	})
+
+	t.Run("Error Reading Request Body", func(t *testing.T) {
+		// Create a test server that sends partial data
+		faultyRequest := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+			w.Header().Set("Content-Length", "10")
+			w.WriteHeader(http.StatusOK)
+			// Write only part of the promised data
+			_, err := w.Write([]byte("Partial "))
+			assert.NoError(t, err)
+
+			// Then flush the writer to send the incomplete response
+			if f, ok := w.(http.Flusher); ok {
+				f.Flush()
+			}
+			// Don't close the connection, just stop writing
+		}))
+		defer faultyRequest.Close()
+
+		// Create a client that will make a request to our test server
+		client := faultyRequest.Client()
+		resp, err := client.Get(faultyRequest.URL)
+		require.NoError(t, err)
+
+		// Use the response body from our test server as the request body for our middleware
+		request := testutil.NewTestRequest(http.MethodPost, "/parse", resp.Body)
+		request.Lang = server.Lang.GetDefault()
+		request.Header().Set("Content-Type", "multipart/form-data")
+
+		result := server.TestMiddleware(&Middleware{}, request, func(resp *goyave.Response, _ *goyave.Request) {
+			resp.Status(http.StatusBadRequest)
```

Comment:
```suggestion
			resp.Status(http.StatusOK)
```
This makes sure the parse middleware doesn't pass because of the response code assertion below. In this case you don't want this line to be executed.

Reply:
Done.